windows/utils/win_install_updates.yml

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Search and install Windows Updates in all categories,
+# and skip optional ones in Windows guest OS.
+# Return:
+#   win_udpates_log_file: Windows Updates install log file path.
+#
+- name: "Set fact of Windows Updates install log path"
+  ansible.builtin.set_fact:
+    win_udpates_log_file: "C:\\win_updates_log.txt"
+
+- name: "Get the list of available Windows Updates"
+  ansible.windows.win_updates:
+    server_selection: "windows_update"
+    category_names: '*'
+    log_path: "{{ win_udpates_log_file }}"
+    skip_optional: true
+    state: "searched"
+  delegate_to: "{{ vm_guest_ip }}"
+  register: win_updates_list
+- name: "Print the list of Windows Updates"
+  ansible.builtin.debug: var=win_updates_list
+
+- name: "Install Windows Updates when updates found"
+  when:
+    - win_updates_list.found_update_count is defined
+    - win_updates_list.found_update_count | int != 0
+    - win_updates_list.updates is defined
+    - win_updates_list.updates | length != 0
+  block:
+    - name: "Install Windows Updates"
+      ansible.windows.win_updates:
+        server_selection: "windows_update"
+        category_names: '*'
+        log_path: "{{ win_udpates_log_file }}"
```

Comment:
Will the output message of installation task overwrite the message of search task?

Reply:
It's ok the content is overwritten by the second task since if the 1st task not failed then the logs are useless.